router_netconsole.c

```diff
@@ -91,7 +156,7 @@ void handle_netconsole_packet(const char *packet_buff, int packet_buff_len,
 				node->router_type->desc);
 		}
 
-		fprintf(stderr, "[%02x:%02x:%02x:%02x:%02x:%02x]: %s router: rebooting.\n",
+		fprintf(stderr, "[%02x:%02x:%02x:%02x:%02x:%02x]: %s router: rebooting (manual reboot might be required).\n",
```

Comment:
Did you try to send the "reset" multiple times? Maybe even with some sleeps between the sends (just for testing purposes - I don't really want to have sleeps in the code)

Reply:
no I did not. but before sending the reset I ensure we have received the command prompt ("u-boot>"), like before sending the flashing command.

However I will try that just for testing and see what happens.

Reply:
Another idea: you can concat multiple commands with some of the u-boot shells. At least with hush you could do something like:

    run fw_upg;reset\n

Just as a different idea which you could test

Reply:
I like this much better! ap51-flash should still be able to understand when the flashing is finished by parsing the output. will give it a try soon

Reply:
it worked! awesome :) I'll add a patch for that!